src/stores/EndpointConfig.tsx

```diff
@@ -0,0 +1,472 @@
+
+    if (errors && errors.length > 0) {
+        response = response.concat(errors);
+    }
+
+    return response;
+};
+
+const filterErrors = (list: JsonFormsData['errors']): (string | undefined)[] =>
+    map(list, 'message');
+
+const populateEndpointConfigErrors = (
+    endpointConfig: JsonFormsData,
+    state: EndpointConfigState,
+    workflow: EntityWorkflow | null
+): void => {
+    const endpointConfigErrors = filterErrors(fetchErrors(endpointConfig));
+
+    state.endpointConfigErrors = endpointConfigErrors.map((message) => ({
+        message,
+    }));
+
+    const editWorkflow =
+        workflow === 'capture_edit' || workflow === 'materialization_edit';
+
+    state.endpointConfigErrorsExist =
+        editWorkflow && !state.serverUpdateRequired
+            ? false
+            : !isEmpty(endpointConfigErrors);
+};
+
+const getInitialStateData = (): Pick<
+    EndpointConfigState,
+    | 'endpointConfig'
+    | 'endpointConfigErrorsExist'
+    | 'endpointConfigErrors'
+    | 'endpointSchema'
+    | 'hydrated'
+    | 'hydrationErrorsExist'
+    | 'publishedEndpointConfig'
+    | 'serverUpdateRequired'
+> => ({
+    endpointConfig: { data: {}, errors: [] },
+    endpointConfigErrorsExist: true,
+    endpointConfigErrors: [],
+    endpointSchema: {},
+    hydrated: false,
+    hydrationErrorsExist: false,
+    publishedEndpointConfig: { data: {}, errors: [] },
+    serverUpdateRequired: false,
+});
+
+const hydrateState = async (
+    get: StoreApi<EndpointConfigState>['getState'],
+    entityType: EntityWithCreateWorkflow,
+    workflow: EntityWorkflow
+): Promise<void> => {
+    const searchParams = new URLSearchParams(window.location.search);
+    const connectorId = searchParams.get(GlobalSearchParams.CONNECTOR_ID);
+    const liveSpecId = searchParams.get(GlobalSearchParams.LIVE_SPEC_ID);
+
+    if (connectorId) {
+        const { data, error } = await getSchema_Endpoint(connectorId);
+
+        if (error) {
+            const { setHydrationErrorsExist } = get();
+
+            setHydrationErrorsExist(true);
+        }
+
+        if (data && data.length > 0) {
+            const { setEndpointSchema } = get();
+
+            setEndpointSchema(
+                data[0].endpoint_spec_schema as unknown as Schema
+            );
+        }
+    }
+
+    if (liveSpecId) {
+        const { data, error } = await getLiveSpecsByLiveSpecId(
+            liveSpecId,
+            entityType
+        );
+
+        if (error) {
+            const { setHydrationErrorsExist } = get();
+
+            setHydrationErrorsExist(true);
+        }
+
+        if (data && data.length > 0) {
+            const { setEndpointConfig, setPublishedEndpointConfig } = get();
+
+            setEndpointConfig(
+                { data: data[0].spec.endpoint.connector.config },
+                workflow
+            );
+
+            setPublishedEndpointConfig({
+                data: data[0].spec.endpoint.connector.config,
+            });
```

Comment:
Where we need to update multiple items in a store with the same data... would it make sense to create specific "helper functions" instead of making multiple calls?